tree/tree/inc/TLeaf.h

```diff
@@ -69,9 +69,14 @@ class TLeaf : public TNamed {
    virtual void     FillBasket(TBuffer &b);
    virtual Int_t   *GenerateOffsetArray(Int_t base, Int_t events) { return GenerateOffsetArrayBase(base, events); }
    TBranch         *GetBranch() const { return fBranch; }
-   virtual TLeaf   *GetLeafCount() const { return fLeafCount; }
+   virtual TLeaf   *GetLeafCount() const { return fLeafCount; } ///< Return a pointer to the leaf that stores the number of elements stored in this leaf. Only meaningful if the leaf contains a variable-length array.
```

Comment:
What about replacing

> the number of elements stored
with
> the size of the variable dimension of the array stored

and

> Only meaningful if the leaf contains a variable-length array.
with
> Returns a nullptr if the leaf does not contain a variable length array.

Reply:
I like the second suggestion -- will put it in asap.

The first suggestion requires the pre-established concept of "variable dimension", which is not documented anywhere. I think I understand from your comments to this PR that ROOT I/O has a concept of "variable length" and "fixed length" of a TLeaf, where the fixed length is always at least 1 and is always well-defined, while the variable length may or may not be defined, and if it is it's stored in another "linked" TLeaf. Notably, a TLeaf can have both a variable length _and_ a fixed length, indicating a multi-dimensional array with some dimensions fixed and some variable, and I have no idea how one then finds out which is which and how much. 

If I got this right and you want to write it somewhere we can point to it from TLeaf's docs, but as things stand I think introducing this in the doxygen comment is too much.

Reply:
Note that the concept is common in C/C++, see for example http://www.ibiblio.org/pub/languages/fortran/append-c.html (search for mat[][3]) where you can specificy the size of the inner dimensions (fixed size) and leave the outer unspecified (variable).

Reply:
I find
```
Return a pointer to the leaf that stores the number of elements stored in this leaf
```
misleading if the type is multi-dimensional (because then the 'number of element stored' is actually GetLenStatic() * valueOf( GetLeafCount ).

Reply:
How about:

```
If this leaf stores a variable-sized array or a multi-dimensional array whose last dimension has variable size, return a pointer to the TLeaf that stores such size. Return a nullptr otherwise.
```

Reply:
Sounds good.  Thanks.